zmq.hpp

```diff
@@ -700,7 +714,7 @@ struct recv_buffer_size
 namespace detail
 {
 
-#ifdef ZMQ_CPP17
+#if ZMQ_HAS_OPTIONAL > 0
```

Comment:
should this not be #ifdef ZMQ_HAS_OPTIONAL? (same below)

Reply:
I believe all preprocessors I know of at least will interpret this correctly, but I suppose
# if defined(ZMQ_HAS_OPTIONAL) && ZMQ_HAS_OPTIONAL>0 would be more correct.  I thought having a value associated with it would allow someone to turn it off even in C++17 mode,  where as just having the test check for definition would not allow that usage.